src/common/details/console/gui/controls/InOutPipeServer.cs

```diff
@@ -0,0 +1,63 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.md file in the project root for full license information.
+//
+
+using System.Text;
+using System.Text.Json;
+
+namespace Azure.AI.Details.Common.CLI.ConsoleGui
+{
+    public class InOutPipeServer
+    {
+        public static bool IsInOutPipeServer => _inOutPipeServer != null;
+
+        public static string? GetInputFromUser(string prompt, string? value = null)
+        {
+            using var stream = new MemoryStream();
+            using var writer = new Utf8JsonWriter(stream, new JsonWriterOptions{ Indented = false });
+            writer.WriteStartObject();
+
+            writer.WriteString("prompt", prompt);
+            if (value != null)
+            {
+                writer.WriteString("value", value);
+            }
+
+            writer.WriteEndObject();
+            writer.Flush();
+
+            var json = Encoding.UTF8.GetString(stream.ToArray());
+            Console.WriteLine($"\n{_inOutPipeServer} GetInputFromUser {json}");
+
+            return Console.ReadLine();
+        }
+
+        public static int GetSelectionFromUser(string[] items, int selected = 0)
+        {
+            using var stream = new MemoryStream();
+            using var writer = new Utf8JsonWriter(stream, new JsonWriterOptions{ Indented = false });
+            writer.WriteStartObject();
+            writer.WriteStartArray("items");
+            foreach (var line in items)
+            {
+                writer.WriteStringValue(line);
+            }
+            writer.WriteEndArray();
+            writer.WriteNumber("selected", selected);
+            writer.WriteEndObject();
+            writer.Flush();
+
+            var json = Encoding.UTF8.GetString(stream.ToArray());
+            Console.WriteLine($"\n{_inOutPipeServer} GetSelectionFromUser {json}");
```

Comment:
This can be simplified to just this single line:
string json = JsonSerializer.Serialize(new { items, selected });

Reply:
Nice catch!